cumulusci/core/tests/test_tasks.py

```diff
@@ -59,6 +59,35 @@ def setUp(self):
         self._task_log_handler.reset()
         self.task_log = self._task_log_handler.messages
 
+    @mock.patch("cumulusci.core.tasks.time.sleep", mock.Mock())
+    def test_retry_on_exception(self):
+        """ calling _retry() should call try until the task succeeds.  """
+        task = BaseTask(self.project_config, self.task_config, self.org_config)
+        task.options = {"retries": 5, "retry_interval": 1, "retry_interval_add": 1}
```

Comment:
It'd be safer to set the task_config's `config` attr before initializing the task...doesn't matter in this case but in general validating options is a thing that happens during task `__init__`

Reply:
i feel like I chose to do this, to make it more clear that the retry logic just uses the compiled task options dict and doesn't rely on any of that setup.